packages/std/src/addresses.rs

```diff
@@ -58,9 +58,33 @@ impl Deref for HumanAddr {
     }
 }
 
+/// Implement `HumanAddr == str`, which gives us `&HumanAddr == &str`.
+/// Do we really need &HumanAddr comparisons?
 impl PartialEq<str> for HumanAddr {
-    fn eq(&self, other: &str) -> bool {
-        self.0 == other
+    fn eq(&self, rhs: &str) -> bool {
+        self.0 == rhs
+    }
+}
+
+/// Implement `str == HumanAddr`, which gives us `&str == &HumanAddr`.
+/// Do we really need &HumanAddr comparisons?
+impl PartialEq<HumanAddr> for str {
+    fn eq(&self, rhs: &HumanAddr) -> bool {
+        self == rhs.0
+    }
+}
```

Comment:
👆 Those two where updated/added for completeness after I realiced you need to implement `A == B` and `B === A` separately. I renamed `other` to `rhs` and added `str == HumanAddr`.